### PR TITLE
fix panic when restart after removeMember

### PR DESCRIPTION
**Pull request by yangxuanjia**
base https://github.com/etcd-io/etcd/pull/13479, merge commit.

### CODE REVIEW

File: server/etcdserver/api/rafthttp/transport.go [hunk 1]
```diff
@@ -339,24 +339,31 @@ func (t *Transport) RemoveAllPeers() {
 
 // the caller of this function must have the peers mutex.
 func (t *Transport) removePeer(id types.ID) {
-	if peer, ok := t.peers[id]; ok {
+	// etcd may remove a member again on startup due to WAL files replaying.
+	// (see: https://github.com/etcd-io/etcd/pull/13479)
```
Comment (ahrtr) [inline server/etcdserver/api/rafthttp/transport.go:343]: The PR number (13479) changes. Suggest to remove the  `// (see: https://github.com/etcd-io/etcd/pull/13479)`.

**Comment by ahrtr** [comment]
Please squash all the commits into one commit.  Please let me know If you don't know how to do it, I am happy to help.

**Comment by yangxuanjia** [comment]
done

**Review by ahrtr — APPROVED** [review]
LGTM

**Comment by serathius** [comment]
cc @ptabor

**Comment by ahrtr** [comment]
@serathius  I just raised a separate ticket to track the task of [adding test ](https://github.com/etcd-io/etcd/issues/13715), I think we can merge this one.

**Review by serathius — APPROVED** [review]

**Comment by serathius** [comment]
ok, let's merge it.

**Pull request merged**